dotnet/src/SemanticKernel.Core/Functions/SemanticFunction.cs

```diff
@@ -204,6 +222,72 @@ private void AddDefaultValues(ContextVariables variables)
         return result;
     }
 
+    /// <summary>
+    /// Handles the FunctionInvoking event
+    /// </summary>
+    /// <param name="context">Execution context</param>
+    /// <param name="renderedPrompt">Rendered prompt</param>
+    private void CallFunctionInvoking(SKContext context, string renderedPrompt)
+    {
+        var eventWrapper = context.FunctionInvokingHandler;
+        if (eventWrapper?.Handler is null)
+        {
+            return;
+        }
+
+        eventWrapper.EventArgs = new FunctionInvokingEventArgs(this.Describe(), context)
+        {
+            Metadata = {
+                [SemanticFunction.RenderedPromptMetadataKey] = renderedPrompt
```

Comment:
Need to think about the discoverability of the metadata. Keys should be defined somewhere more obvious.